presto-iceberg/src/main/java/io/prestosql/iceberg/IcebergPageSourceProvider.java

```diff
@@ -0,0 +1,303 @@
+import static io.prestosql.plugin.hive.HivePageSourceProvider.ColumnMapping.buildColumnMappings;
+import static io.prestosql.plugin.hive.HiveSessionProperties.getParquetMaxReadBlockSize;
+import static io.prestosql.plugin.hive.HiveSessionProperties.isFailOnCorruptedParquetStatistics;
+import static io.prestosql.plugin.hive.parquet.HdfsParquetDataSource.buildHdfsParquetDataSource;
+import static io.prestosql.plugin.hive.parquet.ParquetPageSourceFactory.getParquetTupleDomain;
+import static io.prestosql.plugin.hive.parquet.ParquetPageSourceFactory.getParquetType;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+public class IcebergPageSourceProvider
+        implements ConnectorPageSourceProvider
+{
+    private final HdfsEnvironment hdfsEnvironment;
+    private final TypeManager typeManager;
+    private final FileFormatDataSourceStats fileFormatDataSourceStats;
+
+    @Inject
+    public IcebergPageSourceProvider(
+            HdfsEnvironment hdfsEnvironment,
+            TypeManager typeManager,
+            FileFormatDataSourceStats fileFormatDataSourceStats)
+    {
+        this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        this.fileFormatDataSourceStats = requireNonNull(fileFormatDataSourceStats, "fileFormatDataSourceStats is null");
+    }
+
+    @Override
+    public ConnectorPageSource createPageSource(ConnectorTransactionHandle transaction, ConnectorSession session, ConnectorSplit connectorSplit, ConnectorTableHandle connectorTable, List<ColumnHandle> columns)
+    {
+        IcebergSplit split = (IcebergSplit) connectorSplit;
+        IcebergTableHandle table = (IcebergTableHandle) connectorTable;
+
+        Path path = new Path(split.getPath());
+        long start = split.getStart();
+        long length = split.getLength();
+        List<HiveColumnHandle> hiveColumns = columns.stream()
+                .map(HiveColumnHandle.class::cast)
+                .collect(toList());
+        HdfsContext hdfsContext = new HdfsContext(session, table.getSchemaName(), table.getTableName());
+        return createParquetPageSource(
+                hdfsEnvironment,
+                session.getUser(),
+                hdfsEnvironment.getConfiguration(hdfsContext, path),
+                path,
+                start,
+                length,
+                hiveColumns,
+                split.getNameToId(),
+                false,
+                typeManager,
+                getParquetMaxReadBlockSize(session),
+                isFailOnCorruptedParquetStatistics(session),
+                split.getPredicate(),
+                split.getPartitionKeys(),
+                fileFormatDataSourceStats);
+    }
+
+    private static ConnectorPageSource createParquetPageSource(
```

Comment:
Nit: It looks like it's possible to move the handling of renamed columns out of this method, so that this method can be deprecated and we can reuse `ParquetPageSourceFactory##createPageSource`

Reply:
Added TODO